src/components/Messages/index.vue

```diff
@@ -204,6 +209,18 @@ export default {
     originChanged () {
       this.allowAutoScroll = true
       this.scrolledToMessage = false
+
+      const userIDs = new Set()
+      if (this.origin === 'search') {
+        this.messages.forEach(({ userID }) => {
+          if (!this.findUserByID(userID) && userID) {
+            userIDs.add(userID)
+          }
+        })
+        if (userIDs.size > 0) {
+          this.$store.dispatch('users/fetchUsers', [...userIDs])
+        }
+      }
```

Comment:
`Messages.vue` component is not responsible for this. It should know how to list & show messages, not worry about things such as user existence.

You can define a mixin to handle this for Channel, Thread, ...